stack-graphs/src/graph.rs

```diff
@@ -1408,6 +1409,155 @@ impl StackGraph {
     pub fn new() -> StackGraph {
         StackGraph::default()
     }
+
+    /// Copies the given stack graph into this stack graph. Panics if any of the files
+    /// in the other stack graph are already defined in the current one.
+    pub fn add_graph(&mut self, other: &StackGraph) -> Result<(), Handle<File>> {
```

Comment:
I don’t love this name. Maybe `copy_graph` or `populate_from_graph`? Also, is there a reason you chose `Result<(), Handle<File>>` over `Option<Handle<File>>`?

Reply:
I picked this one in line with `add_edge`, `add_node` etc. If you think that this is something else than those, I think `populate_from_graph` is a fine name.

Reply:
I think it conveys the meaning better. Using `Option`, the success case would actually be `None`, which is not typically how `Option` is used. Does that make sense?

Reply:
How about `add_from_graph`? And my bad re. the `Result`, your explanation makes sense; I forgot that Rust reorders its parameters compared to Haskell’s `Either`.

Reply:
Like the `add_from_graph`!